sentry-android-core/src/main/java/io/sentry/android/core/PerformanceAndroidEventProcessor.java

```diff
@@ -41,6 +48,16 @@
       }
     }
 
+    final SentryId eventId = transaction.getEventId();
+    if (eventId != null) {
+      final Map<String, @NotNull MeasurementValue> framesMetrics =
+          ActivityFramesTracker.getInstance().getMetrics(eventId);
```

Comment:
Could we replace the `getMetrics` and `removeMetrics` with a single `takeMetrics` (or a better name) that would do both?

Reply:
yes, thats possible